pysph/sph/boundary/simple_inlet_outlet.py

```diff
@@ -0,0 +1,86 @@
+"""Simple Inlet and Outlet support for PySPH.  The inlet and outlets are axis
+aligned.
+
+The Inlet lets a user stack a given set of inlet particles along a particular
+coordinate axis.  As the particles move they are copied over to a specified
+destination particle array.  The particles are also wrapped back into the
+inlet.
+
+The Outlet lets a user specify a region where any particles enter will be
+added to a specified "outlet" particle array and removed from the source
+array.  When outlet particles leave the outlet region (specified with a
+bounding box) they are removed from the simulation.
+
+"""
+# Copyright (c) 2015-2017, Prabhu Ramachandran
+# License: BSD
+
+from pysph.sph.integrator_step import IntegratorStep
+from pysph.sph.integrator import PECIntegrator
+from pysph.sph.scheme import TVFScheme
+from pysph.sph.wc.edac import EDACScheme
+from pysph.sph.boundary.base import InletOutletManager
+
+
+class SimpleInletOutlet(InletOutletManager):
+    def add_io_properties(self, pa, scheme=None):
+        DEFAULT_PROPS = ['xn', 'yn', 'zn', 'ioid', 'disp',
+                         'wij']
+        for prop in DEFAULT_PROPS:
+            pa.add_property(prop)
+
+    def get_stepper(self, scheme, cls):
+        steppers = {}
+        if (cls == PECIntegrator):
+            if isinstance(scheme, EDACScheme):
+                for inlet in self.inlets:
+                    steppers[inlet] = InletOutletStep_edac()
+                for outlet in self.outlets:
+                    steppers[outlet] = InletOutletStep_edac()
+                self.active_stages = [2]
+
+            elif isinstance(scheme, TVFScheme):
+                for inlet in self.inlets:
+                    steppers[inlet] = InletOutletStep_tvf()
+                for outlet in self.outlets:
+                    steppers[outlet] = InletOutletStep_tvf()
+                self.active_stages = [2]
+
+        return steppers
+
+    def get_equations(self, scheme=None, summation_density=False):
+        from pysph.sph.equation import Group
+
+        equations = []
+        g00 = []
+        for inlet in self.inletinfo:
+            for eqn in inlet.equations:
+                g00.append(eqn)
+        for outlet in self.outletinfo:
+            for eqn in outlet.equations:
+                g00.append(eqn)
+
+        if g00:
+            equations.append(Group(equations=g00, real=False))
+
+        return equations
+
+
+class InletOutletStep_tvf(IntegratorStep):
```

Comment:
Use CamelCase so make this: `InletOutletStepTVF`.

Reply:
done